File: examples/nodejs-agent/check-balance.ts
/**
 * Minimal example — check node status and wallet balance.
 *
 * Usage:
 *   node --loader ts-node/esm check-balance.ts
 */

import { ClawNetClient } from '@claw-network/sdk';

const client = new ClawNetClient();

const status = await client.node.getStatus();
console.log('Node:', status.network, `block #${status.blockHeight}`, status.synced ? '(synced)' : '(syncing …)');

const balance = await client.wallet.getBalance();
console.log(`Balance : ${balance.balance} Tokens`);
console.log(`  available : ${balance.available}`);
console.log(`  pending   : ${balance.pending}`);
console.log(`  locked    : ${balance.locked}`);

File: examples/python-agent/check_balance.py
#!/usr/bin/env python3
"""Quick helper: check wallet balance from the command line."""

from __future__ import annotations

import os
import sys

from clawnet import ClawNetClient, ClawNetError

NODE_URL = os.environ.get("CLAW_NODE_URL", "http://127.0.0.1:9528")


def main() -> None:
    did = sys.argv[1] if len(sys.argv) > 1 else None
    client = ClawNetClient(NODE_URL)

    try:
        balance = client.wallet.get_balance(did)
        print(f"Available : {balance['available']} CLAW")
        print(f"Locked    : {balance['locked']} CLAW")
        print(f"Total     : {balance['total']} CLAW")
    except ClawNetError as exc:
        print(f"Error ({exc.status}): {exc}", file=sys.stderr)
        sys.exit(1)


if __name__ == "__main__":
    main()
